dev/bots/test.dart

```diff
@@ -1121,7 +1121,9 @@ String get gitHash {
 /// Returns null if the contents are good. Returns a string if they are bad.
 /// The string is an error message.
 Future<String> verifyVersion(File file) async {
-  final RegExp pattern = RegExp(r'^\d+\.\d+\.\d+(\+hotfix\.\d+)?(-pre\.\d+)?$');
+  // 1.18.0-5.0-pre.8
+  final RegExp pattern = RegExp(
+    r'^(\d+)\.(\d+)\.(\d+)(-\d+\.\d+)?(-pre\.\d+)?$');
```

Comment:
`.pre` (dot, not hyphen).

And elsewhere.

Reply:
I can make that change, but to clarify, this is just for reporting commits past last tagged version, and not the git tag format itself, and is a break from how we used to do it.

```
// old way, 4 commits past 1.2.3
1.2.4-pre.4

// this PR, 4 commits past 1.2.3-0.0.pre
1.2.3-1.0-pre.4

// proposed change, 4 commits past 1.2.3-0.0.pre
1.2.3-1.0.pre.4
```

Reply:
and also, if you are 4 commits past version 1.2.3? Should that be `1.2.3-pre.4` or `1.2.3.pre.4`?